resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/SparkKubernetesClientFactory.scala

```diff
@@ -31,13 +36,64 @@ import org.apache.spark.internal.Logging
 import org.apache.spark.internal.config.ConfigEntry
 import org.apache.spark.util.ThreadUtils
 
+private[spark] class SparkOAuthTokenProvider(config: File) extends OAuthTokenProvider {
+  val kubeConfig = KubeConfig.loadKubeConfig(new FileReader(config))
+  val persister = new FilePersister(config)
+  kubeConfig.setPersistConfig(persister)
+
+  def getToken(): String = {
+    return kubeConfig.getAccessToken()
+  }
+}
+
 /**
  * Spark-opinionated builder for Kubernetes clients. It uses a prefix plus common suffixes to
  * parse configuration keys, similar to the manner in which Spark's SecurityManager parses SSL
  * options for different components.
  */
 private[spark] object SparkKubernetesClientFactory extends Logging {
 
+   /**
+   * Check if the code is being run from within kubernetes.
+   * @return
+   */
+  def isOnKubernetes(): Boolean = {
+    val serviceHost = System.getenv("KUBERNETES_SERVICE_HOST")
+    return serviceHost != null && serviceHost.length > 0
+  }
+
+    def getHomeDir(): String = {
```

Comment:
Where's this coming from?

Reply:
I meant more are you writing this code from scratch or are you porting it from somewhere?

Reply:
Oh it's coming from @daggertheog's [diff](https://github.com/Affirm/spark/commit/40bd78e44335fe9ef481b0d9461c05b1d72aad0f) for the spark 2 k8s auth issue